testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -8815,6 +8815,10 @@ private static Consumer<List<? extends Throwable>> checkWarnings(
         .ok("JSON_OBJECT(KEY 'foo' VALUE "
             + "JSON_OBJECT(KEY 'foo' VALUE 'bar' NULL ON NULL) "
             + "FORMAT JSON NULL ON NULL)");
+    expr("json_object('foo', 'bar')")
```

Comment:
how about some other tests, e.g., with an odd number of elements?

Reply:
How about mixed syntax tests? e.g.
```
json_object('foo': json_object('foo', 'bar') format json)
```